apps/lex-core-service/src/main/java/com/infosys/lex/rating/service/RatingServiceImpl.java

```diff
@@ -134,23 +139,91 @@
 	}
 	
 	@Override
-	public Map<String,Object> getRatingsInfoForContents(String rootOrg,ContentIdsDto contentIdsMap)
-	{
-		Map<String,Object> resp = new HashMap<>();
+	public Map<String,Object> getRatingsInfoForContents(String rootOrg,ContentIdsDto contentIdsMap) {
+		Map<String, Object> resp = new HashMap<>();
 		String contentId;
-		
+
 		List<String> contentIds = contentIdsMap.getContentIds();
-		List<Map<String,Object>> ratingsInfoList = userResourceRatingRepo.getAvgRatingAndRatingCountForContentIds(rootOrg, contentIds);
-		
-		for(Map<String,Object> ratingsInfo : ratingsInfoList)
-		{
+		List<Map<String, Object>> ratingsInfoList = userResourceRatingRepo.getAvgRatingAndRatingCountForContentIds(rootOrg, contentIds);
+
+		List<Integer> defaultRating = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
+		HashMap<String, Object> ratingValue;
+		ArrayList<Object> ratingInfo = new ArrayList<>();
+		for (Map<String, Object> ratingsInfo : ratingsInfoList) {
 			contentId = ratingsInfo.get("content_id").toString();
 			ratingsInfo.remove("content_id");
-			ratingsInfo.put("averageRating",df.format((float)ratingsInfo.get("averageRating") ));
+			ratingsInfo.put("averageRating", df.format((float) ratingsInfo.get("averageRating")));
+
+			List<Float> ratings = userResourceRatingRepo.getRatingsForContent(rootOrg, contentId);
+			Map<Float, Long> streamResult = ratings.stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+			for (Integer defaultValue : defaultRating) {
+				ratingValue = new HashMap<>(2);
+				ratingValue.put("rating", defaultValue);
+				ratingValue.put("count", streamResult.getOrDefault(Float.valueOf(defaultValue), 0L));
+				ratingInfo.add(ratingValue);
+			}
+			ratingsInfo.put("ratingInfo", ratingInfo);
+
 			resp.put(contentId, ratingsInfo);
 		}
 		return resp;
-		
+
+	}
+
+	/**
+	 *
+	 * @param rootOrg root org
+	 * @param contentId content Id
+	 * @return return rating info for a content Id
+	 */
+	@Override
+	public Map<String, Object> getRatingsInfoForContents(String rootOrg, String contentId) {
```

Comment:
Map<String, Object> getRatingsInfoForContents(String rootOrg, ContentIdsDto contentIdsMap);
Map<String, Object> getRatingsInfoForContents(String rootOrg, String contentId);

The above two method implementation repeats plenty of code... create a common method and provide the implementation.